lib/dragonrealms/commons/common-arcana.rb

```diff
@@ -497,20 +497,11 @@ def release_cyclics(cyclic_no_release = [])
       def parse_regalia # generates an array of currently-worn regalia armor nouns
         return unless DRStats.trader?
 
-        DRC.bput('inv armor', 'Use INVENTORY HELP for more options')
-        snapshot = reget(40)
-        if snapshot.grep(/All of your armor|You aren't wearing anything like that/).any? && snapshot.grep(/Use INVENTORY HELP/).any?
-          snapshot
-            .map(&:strip)
-            .reverse
-            .take_while { |item| !['All of your armor:', "You aren't wearing anything like that."].include?(item) }
-            .drop_while { |item| item != '[Use INVENTORY HELP for more options.]' }
-            .drop(1)
-            .select { |item| item.include?('rough-cut crystal') || item.include?('faceted crystal') || item.include?('resplendent crystal') }
-            .map { |item| DRC.get_noun(item) }
-        else
-          parse_regalia
-        end
+        snapshot = Lich::Util.issue_command("inv combat", /All of your worn combat|You aren't wearing anything like that/, /Use INVENTORY HELP for more options/, usexml: false, include_end: false)
+                             .map(&:strip)
+        snapshot = snapshot - ["All of your worn combat equipment:", "You aren't wearing anything like that."]
+        snapshot.select { |item| item.include?('rough-cut crystal') || item.include?('faceted crystal') || item.include?('resplendent crystal') }
```

Comment:
```suggestion
        (snapshot - ["All of your worn combat equipment:", "You aren't wearing anything like that."]).select { |item| item.include?('rough-cut crystal') || item.include?('faceted crystal') || item.include?('resplendent crystal') }
```

Can simplify a hair here.